### PR TITLE
docs: Add example to Server Side Calls for Next.js API

### PR DESCRIPTION
**Pull request by bennettdams**
## 🎯 Changes

This adds an example on how Server Side Calls could look like in a custom Next.js API endpoint.
It is also referenced in the "Error Handling" docs, where `getHTTPStatusCodeFromError` is shown.

The idea to add this example is coming from `getHTTPStatusCodeFromError` [now being public](https://github.com/trpc/trpc/pull/3045).

I thought about adding such endpoint in the real code examples, but I don't know which of them would be appropriate, if any (maybe `next-prisma-starter`?). 

And also, maybe people who are new to tRPC get the wrong impression of how to use tRPC in general, as Server Side Calls shouldn't be the default way of using tRPC with Next.js.

---

I tried out the example code in a real Next.js API endpoint btw, works fine:

image

## ✅ Checklist

- [x] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/main/CONTRIBUTING.md).
- [x] If necessary, I have added documentation related to the changes made.
- [ ] I have added or updated the tests related to the changes made.

### CODE REVIEW

File: www/docs/server/server-side-calls.md [hunk 1]
```diff
@@ -129,3 +129,60 @@ const router = t.router({
...
+This example shows how to use the caller in a Next.js API endpoint. tRPC creates API endpoints for you already, so this file is only meant to show
+how to call a procedure from another, custom endpoint.
+
+:::
+
+```ts twoslash
+import { TRPCError } from '@trpc/server';
+import { getHTTPStatusCodeFromError } from '@trpc/server/http';
+import type { NextApiRequest, NextApiResponse } from 'next';
+import { appRouter } from '~/server/routers/_app';
```
Comment (juliusmarminge) [inline www/docs/server/server-side-calls.md:146]: either create a router (can be hidden) or add a noErrors directive

File: www/docs/server/server-side-calls.md [hunk 2]
```diff
@@ -129,3 +129,60 @@ const router = t.router({
...
+### Example for a Next.js API endpoint
+
+:::tip
+
+This example shows how to use the caller in a Next.js API endpoint. tRPC creates API endpoints for you already, so this file is only meant to show
+how to call a procedure from another, custom endpoint.
+
+:::
+
+```ts twoslash
```
Comment (juliusmarminge) [inline www/docs/server/server-side-calls.md:142]: just silence errors
```suggestion
```ts twoslash
// @noErrors
// ---cut---
```

or create a router if you want to show types etc

**Comment by bennettdams** [comment]
> Build fails :(

Build succeeds now :) Thanks, @juliusmarminge.

**Review by KATT — DISMISSED** [review]

**Comment by bennettdams** [comment]
@KATT You renamed the caught variable to `cause`, so I now [updated the relying code](https://github.com/trpc/trpc/pull/3051/commits/7516ee1bf926236db0be49fa7f18df8bdb3c2a44) as well 😅

File: www/docs/server/server-side-calls.md [hunk 3]
```diff
@@ -171,13 +171,12 @@ export default async (
...
   } catch (cause) {
-
     // If this a tRPC error, we can extract additional information.
     if (cause instanceof TRPCError) {
       // We can get the specific HTTP status code coming from tRPC (e.g. 404 for `NOT_FOUND`).
-      const httpStatusCode = getHTTPStatusCodeFromError(error);
+      const httpStatusCode = getHTTPStatusCodeFromError(cause);
 
-      res.status(httpStatusCode).json({ error: { message: error.message } });
+      res.status(httpStatusCode).json({ error: { message: cause.message } });
```
Comment (KATT) [inline www/docs/server/server-side-calls.md:179]: This `cause` might not actually be an `Error`

`throw 'wat'` is valid

Comment (bennettdams) [reply www/docs/server/server-side-calls.md:179]: We're inside the `instanceof TRPCError` block, so `message` is always set, no?
Are you saying it should also return the `cause` directly, e.g. `message: cause.message ?? cause`?

File: examples/next-prisma-starter-websockets/src/server/routers/_app.ts [hunk 4]
```diff
@@ -7,9 +7,7 @@ import { observable } from '@trpc/server/observable';
 import { clearInterval } from 'timers';
 
 export const appRouter = router({
-  healthcheck: publicProcedure.query(() => {
```
Comment (juliusmarminge) [inline examples/next-prisma-starter-websockets/src/server/routers/_app.ts:10]: Sorry pushed to the wrong branch 😩 Will revert tomorrow if you havent already

File: www/docs/server/server-side-calls.md (hunk 3 above)
Comment (bennettdams) [reply www/docs/server/server-side-calls.md:176]: @KATT Maybe you missed this?

**Review by KATT — APPROVED** [review]

**Pull request merged**